src/store/name-generator/reducers/name-generator.reducers.ts

```diff
@@ -22,5 +25,15 @@ export const nameGeneratorReducer = createReducer(initialState)
     loading: false,
     errors: action.payload,
   }))
-  .handleAction('@name-generator/SET_BOY', (state: NameGeneratorState) => ({ ...state, boyId: state.boyId + 1 }))
-  .handleAction('@name-generator/SET_GIRL', (state: NameGeneratorState) => ({ ...state, girlId: state.girlId + 1 }));
+  .handleAction('@name-generator/SET_BOY', (state: NameGeneratorState) => {
+    let currentIndex: number = state.boyId;
+    let lastIndex: number = state.petNames.boys.length-1;
+    let nextIndex: number = currentIndex === lastIndex ? 0 : currentIndex + 1; 
```

Comment:
I see that these variables have no re-assignments in the code, so that you can freely use `const` instead of `let`.

Reply:
nit: It is not require to specif `: number` type for the immediate assignment.

Not perfect
```JavaScript
const fruit: string = 'apple'; 
```
Perfect
```JavaScript
const fruit = 'apple'; 
```

Reply:
done, thanks